crates/symbolicator/src/services/download/s3.rs

```diff
@@ -120,8 +146,9 @@ impl S3Downloader {
         log::debug!("Fetching from s3: {} (from {})", &key, &bucket);
 
         let source_key = &file_source.source.source_key;
+        let aws_credentials_provider = &source_key.aws_credentials_provider;
         let result = self
-            .get_s3_client(&source_key)
+            .get_s3_client(&aws_credentials_provider.to_string(), &source_key)
```

Comment:
Since the provider is another property of the `source_key`, you don’t need to pass it in as a separate argument. Also, you do a `to_string` here and immediately taking a reference, which is needlessly allocating.

Reply:
Fixed.